lib/finch.ex

```diff
@@ -399,9 +399,12 @@ defmodule Finch do
     * `:pool_timeout` - This timeout is applied when we check out a connection from the pool.
       Default value is `5_000`.
 
-    * `:receive_timeout` - The maximum time to wait for a response before returning an error.
+    * `:receive_timeout` - The maximum time to wait for each chunk to be received before returning an error.
       Default value is `15_000`.
 
+    * `:request_timeout` - The maximum time to wait for a complete response before returning an error.
```

Comment:
I think we should mention how this is a best effort timeout and that the current implementation does not guarantee that the call will return before this amount of time passes. We should also mention that this has only been implemented for HTTP/1.